tests/system_tests_alternate_dest.py

```diff
@@ -0,0 +1,781 @@
+
+    def on_link_opened(self, event):
+        if event.sender == self.sender:
+            self.receiver = event.container.create_receiver(self.receiver_conn, self.addr)
+            if self.rx_alternate:
+                self.receiver.source.capabilities.put_symbol("qd.alternate")
+
+    def on_sendable(self, event):
+        if event.sender == self.sender:
+            while self.sender.credit > 0 and self.n_tx < self.count:
+                self.sender.send(Message("Message %d" % self.n_tx))
+                self.n_tx += 1
+
+    def on_message(self, event):
+        if event.receiver == self.receiver:
+            self.n_rx += 1
+            if self.n_rx == self.count:
+                self.fail(None)
+
+    def on_released(self, event):
+        self.n_rel += 1
+
+    def run(self):
+        Container(self).run()
+
+
+class ReceiverFirstTest(MessagingHandler):
+    def __init__(self, sender_host, receiver_host, addr, rx_alternate):
+        super(ReceiverFirstTest, self).__init__()
+        self.sender_host   = sender_host
+        self.receiver_host = receiver_host
+        self.addr          = addr
+        self.rx_alternate  = rx_alternate
+        self.count         = 300
+
+        self.sender_conn   = None
+        self.receiver_conn = None
+        self.error         = None
+        self.n_tx          = 0
+        self.n_rx          = 0
+        self.n_rel         = 0
+
+    def timeout(self):
+        self.error = "Timeout Expired - n_tx=%d, n_rx=%d, n_rel=%d" % (self.n_tx, self.n_rx, self.n_rel)
+        self.sender_conn.close()
+        self.receiver_conn.close()
+
+    def fail(self, error):
+        self.error = error
+        self.sender_conn.close()
+        self.receiver_conn.close()
+        self.timer.cancel()
+
+    def on_start(self, event):
+        self.timer         = event.reactor.schedule(10.0, Timeout(self))
+        self.sender_conn   = event.container.connect(self.sender_host)
+        self.receiver_conn = event.container.connect(self.receiver_host)
+        self.receiver      = event.container.create_receiver(self.receiver_conn, self.addr)
+        if self.rx_alternate:
+            self.receiver.source.capabilities.put_symbol("qd.alternate")
+
+    def on_link_opened(self, event):
+        if event.receiver == self.receiver:
+            self.sender = event.container.create_sender(self.sender_conn, self.addr)
+
+    def on_sendable(self, event):
+        if event.sender == self.sender:
+            while self.sender.credit > 0 and self.n_tx < self.count:
+                self.sender.send(Message("Message %d" % self.n_tx))
+                self.n_tx += 1
+
+    def on_message(self, event):
+        if event.receiver == self.receiver:
+            self.n_rx += 1
+            if self.n_rx == self.count:
+                self.fail(None)
+
+    def on_released(self, event):
+        self.n_rel += 1
+
+    def run(self):
+        Container(self).run()
+
+
+class SwitchoverTest(MessagingHandler):
```

Comment:
A couple of additions to the SwitchoverTest
1. Close the alternate_receiver also. Now since the messages have nowhere to go, do they get released
2. Close the alternate_receiver and bring the primary_receiver back up again to see if the messages start flowing to the primary_receiver
3. Dont close the alternate_receiver and bring back the primary_receiver to make sure the primary_receiver starts receiving messages.

Apply the above test cases to the auto link tests as well.